lib/src/form_builder_validators.dart

```diff
@@ -201,25 +201,28 @@ class FormBuilderValidators {
     };
   }
 
+  // TODO(any): l10n
   /// [FormFieldValidator] that requires the field's value to be a valid integer.
-  static FormFieldValidator integer({
+  static FormFieldValidator integer(
+    BuildContext context, {
     String errorText = 'Value must be an integer.',
   }) {
```

Comment:
For maximum flexibility, consider adding an optional `radix` parameter (which `int.tryParse` accepts).

Reply:
You are right, to be a little more complete, it is also necessary to add an optional `positive` parameter (for integer and double)

Reply:
Be careful about overloading too many ideas into one validator because it can easily complicate the error messaging. You ideally want _ONE_ discrete idea to communicate in an error message, not a complex _set_ of ideas.

The `compose` factory lets you combine multiple validation ideas (like required, numeric, and positive).

For enforcing Positive values, I would probably add a `min` constraint.

Reply:
This is totally right.

Reply:
I think it is relevant to add

```dart
/// [FormFieldValidator] that requires the field's value to be a positive number.
static FormFieldValidator positive(
  BuildContext context, {
  String errorText = 'Value must be a positive number.',
}) {
  return min(context, 0, errorText: errorText);
}
```
(I noticed that the minimum validator also checks that the value is not null (and therefore it is required).)

Reply:
This version from `master` will not error when `valueCandidate` is empty because `num.tryParse` will return _null_.

```dart
  /// [FormFieldValidator] that requires the field's value to be greater than
  /// or equal to the provided number.
  static FormFieldValidator min(
    num min, {
    String errorText,
  }) {
    return (valueCandidate) {
      if (valueCandidate != null &&
          ((valueCandidate is num && valueCandidate < min) ||
              (valueCandidate is String &&
                  num.tryParse(valueCandidate) != null &&
                  num.tryParse(valueCandidate) < min))) {
        return errorText ?? 'Value must be greater than or equal to $min';
      }
      return null;
    };
  }
```

Reply:
Yes I misread the code 🤦‍♂️

Reply:
I'm happy to yield to @danvick but my recommendation is to not add a `positive` validator because it sort of begs for an equally opposite, but not as useful, `negative` validator.  Also, this creates 2 strategies to do something (either use the `positive` validator, or the `min` validator) -- _less is more._

There is a lot of opportunity left with validators, like:
 * Is the Min/Max inclusive (>=/<=) or exclusive (>/<)?
 * Can we get a MinRange/MaxRange -- where the min field must be less than the max field, and the max field must be greater than the min field.  (Tricky part is providing a _getter_ for the _other field_.)
 * Two fields match, or don't match -- useful for password confirmation screens.

I'm getting carried away here.  I think I'll walk away and wait for an update.

Reply:
> Yes I misread the code 🤦‍♂️

Not your fault -- that code is too hard to read!  (I would rewrite that to make it easier to comprehend.)

Reply:
> > Yes I misread the code 🤦‍♂️
> 
> Not your fault -- that code is too hard to read! (I would rewrite that to make it easier to comprehend.)

See https://github.com/danvick/flutter_form_builder/pull/518#discussion_r515897768